analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicValue.cs

```diff
@@ -25,11 +25,12 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn
     public sealed record SymbolicValue
     {
         // Reuse instances to save memory. This "True" has the same semantic meaning and any other symbolic value with BoolConstraint.True constraint
-        public static readonly SymbolicValue This = new SymbolicValue().WithConstraint(ObjectConstraint.NotNull);
-        public static readonly SymbolicValue Null = new SymbolicValue().WithConstraint(ObjectConstraint.Null);
-        public static readonly SymbolicValue NotNull = new SymbolicValue().WithConstraint(ObjectConstraint.NotNull);
-        public static readonly SymbolicValue True = new SymbolicValue().WithConstraint(BoolConstraint.True);
-        public static readonly SymbolicValue False = new SymbolicValue().WithConstraint(BoolConstraint.False);
+        public static readonly SymbolicValue Constraintless = new();
```

Comment:
A first simple step for the "Avoid allocations in ProgramState.SetOperationConstraint" sub-task. The next step would be to make the SymbolicValue constructor private (this only leads to simple changes in the UTs). Alternative names for `Constraintless` I considered are:
* Default
* Empty
* NoConstraints